pkg/cache/v2/cluster_test.go

```diff
@@ -171,3 +178,177 @@ func TestClusterFlush(t *testing.T) {
+
+	cluster := cluster_v2.Cluster{
+		ApiStatus:      core_v2.ApiStatus_UPDATE,
+		ConnectTimeout: uint32(1),
+		CircuitBreakers: &cluster_v2.CircuitBreakers{
+			MaxConnections:     uint32(4294967295),
+			MaxPendingRequests: uint32(4294967295),
+			MaxRequests:        uint32(4294967295),
+			MaxRetries:         uint32(4294967295),
+		},
+		LoadAssignment: &endpoint.ClusterLoadAssignment{
+			ClusterName: "inbound|9080|http|reviews.default.svc.cluster.local",
+			Endpoints: []*endpoint.LocalityLbEndpoints{
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9090),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.240"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9091),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.241"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9092),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.242"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9293),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.243"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9294),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.244"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9095),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.245"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9096),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.246"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9097),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.247"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9098),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.248"),
+							},
+						},
+					},
+				},
+				{
+					LbEndpoints: []*endpoint.Endpoint{
+						{
+							Address: &core_v2.SocketAddress{
+								Port: uint32(9099),
+								Ipv4: nets.ConvertIpToUint32("192.168.127.249"),
+							},
+						},
+					},
+				},
+			},
+		},
+		LbPolicy: cluster_v2.Cluster_ROUND_ROBIN,
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		cache := NewClusterCache()
+		cluster.Name = rand.String(6)
+		cache.SetApiCluster(cluster.Name, &cluster)
+
+		cache.Flush()
```

Comment:
You didn't set the api status, actually the flush do nothing

Reply:
I set Cluster.ApiStatus in L229. and compared the value of api status after Flush() processing. 
Its changed from `update` to `none`, which shows that Flush did the processing.